### PR TITLE
Support for code coverage

### PR DESCRIPTION
**Pull request by levmckinney**
Adds support for code coverage to the package. Ideally this should enforce 100% coverage on tests and increasing coverage for all subsequent pull requests.

### CODE REVIEW

**Comment by norabelrose** [comment]
Is this like a ratcheting thing where it never lets the coverage go down in a new PR? I agree we need more coverage than we currently do although I remember `codecov` was sort of a pain in the ass when contributing to `imitation`

**Comment by levmckinney** [comment]
> Is this like a ratcheting thing where it never lets the coverage go down in a new PR? I agree we need more coverage than we currently do although I remember `codecov` was sort of a pain in the ass when contributing to `imitation`

At the moment all it does is make sure that overall code coverage does not go down. I'm inclined to configure it to require X% code coverage for all new lines added by a pull requests. I'm thinking like 95%, so you don't have to worry about covering every error, but most paths are getting covered.

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    # Don't complain if tests don't hit defensive assertion code:
+    raise AssertionError
+    raise NotImplementedError
+
+    # Don't complain if non-runnable code isn't run:
+    if 0:
+    if __name__ == .__main__.:
+
+    # Don't complain about abstract methods, they aren't run:
```
Comment (AdamGleave) [inline .coveragerc:24]: Not necessarily true, abstract methods can have concrete implementations and get called by `super()`. Maybe exclude `pass` and `...` instead that are used as placeholders for abstract methods?

File: docs/source/_api/tuned_lens.load_artifacts.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿.. _tuned\_lens.load\_artifacts:
```
Comment (AdamGleave) [inline docs/source/_api/tuned_lens.load_artifacts.rst:1]: Do these docs belong in this PR?

**Review by AdamGleave — APPROVED** [review]
LGTM although would still suggest changing handling of abstract methods to ignore the pass stub rather than all abstract methods (which may include concrete code that does get executed).

In future please avoid force-pushing in PRs as it makes it harder to review the diff.

**Pull request merged**